tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,260 @@
+           Please add the consent to Config.xml or delete columns and data from 'participant_status'"
+        );
+    }
+}
+// Check participant_status columns and data for each consent in Config.xml
+foreach ($consentList as $consentName=>$consentLabel) {
+    $consentDate       = $consentName . "_date";
+    $consentWithdrawal = $consentName . "_withdrawal";
+    $statusExists      = $db->columnExists('participant_status', $consentName);
+    $dateExists        = $db->columnExists('participant_status', $consentDate);
+    $withdrawalExists  = $db->columnExists('participant_status', $consentWithdrawal);
+    // Check that no column is missing
+    if (!($statusExists && $dateExists && $withdrawalExists)) {
+        array_push($errors, "At least one column is missing for " . $consentName . " in participant_status table.
+           Check that status, date, and withdrawal columns exist for this consent type.");
+    }
+    // Check data
+    $dataQuery = "SELECT ID, CandID, " . $db->escape($consentName) . ", " . $db->escape($consentDate) . ", " .
+                 $db->escape($consentWithdrawal) . " FROM participant_status";
+    $psData    = $db->pselect($dataQuery, array());
+    foreach ($psData as $entry) {
+        $status     = $entry[$consentName];
+        $date       = $entry[$consentDate];
+        $withdrawal = $entry[$consentWithdrawa];
+        $consentID  = $entry['ID'];
+        $candID     = $entry['CandID'];
+        // Check if consent status is not_answered
+        if($status === "not_answered") {
+            array_push($errors, "Deprecated consent status 'not_answered' found for
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       Please change to a valid status or remove data.");
+        }
+        // Check if consent status is null or empty but there are dates
+        if(empty($status) && (!empty($date) || !empty($withdrawal))) {
+            array_push($errors, "Consent status not found for
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . " where there are dates.");
+        }
+        // Check if consent status is yes but consent date is empty
+        if($status === "yes" && empty($date)) {
+            array_push($errors, "The date of consent is missing for
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       All 'yes' statuses must have a consent date.");
+        }
+        // Check if consent status is no but withdrawal date is empty
+        if($status === "no" && empty($withdrawal)) {
+            array_push($errors, "The date of withdrawal is missing for 
+                       [ID]     => " . $consentID . "
+                       [CandID] => " . $candID . "
+                       All 'no' statuses must have a withdrawal date.");
```

Comment:
On IBIS, there are a ton of candidates who have `study_consent` set to "no" and `study_consent_date` and `study_consent_withdrawal` both set to NULL. I think your code should first check whether there was a `study_consent_date` ever set before requiring a `study_withdrawal_date` to have a value other than NULL.